src/Databases/DatabasesCommon.cpp

```diff
@@ -20,6 +22,34 @@ namespace ErrorCodes
     extern const int UNKNOWN_DATABASE;
 }
 
+/// Daisy: starts.
+std::tuple<String, String> IDatabase::astToQueryAndEngineStringImpl(ASTPtr ast, ContextPtr context) const
+{
+    String query_string;
+    String engine_full;
+    if (ast)
+    {
+        auto & ast_create = ast->as<ASTCreateQuery &>();
+        if (!context->getSettingsRef().show_table_uuid_in_table_create_query_if_not_nil)
+        {
+            ast_create.uuid = UUIDHelpers::Nil;
+            ast_create.to_inner_uuid = UUIDHelpers::Nil;
+        }
+        query_string = queryToString(ast);
+
+        if (ast_create.storage)
```

Comment:
If ast is not a ASTCreateQuery, here it throws exception. Before access .storage, make sure ast_create is not null

Reply:
here `auto & ast_create = ast->as<ASTCreateQuery &>()` , if cast to **&** , only maybe throw exception.